modules/standard/Random.chpl

```diff
@@ -69,115 +68,152 @@ module Random {
 
-    :arg seed: The seed to use for the PRNG.  Defaults to
-     `oddCurrentTime` from :type:`RandomSupport.SeedGenerator`.
-    :type seed: `int(64)`
+    :arg arr: An array of numeric values
 
-    :arg algorithm: A param indicating which algorithm to use. Defaults to :param:`defaultRNG`.
-    :type algorithm: :type:`RNG`
   */
-  proc fillRandom(ref arr: [], seed: int(64) = SeedGenerator.oddCurrentTime, param
-      algorithm=defaultRNG)
-    where isSupportedNumericType(arr.eltType) {
-    var randNums = createRandomStream(seed=seed,
-                                      eltType=arr.eltType,
-                                      parSafe=false,
-                                      algorithm=algorithm);
-    randNums.fillRandom(arr);
+  proc fillRandom(ref arr: [] ?t)
+    where isNumericOrBoolType(t) && arr.isRectangular()
+  {
+    var rs = new randomStream(t);
+    rs.fill(arr);
   }
 
   @chpldoc.nodoc
-  proc fillRandom(ref arr: [], seed: int(64) = SeedGenerator.oddCurrentTime, param
-      algorithm=defaultRNG) {
-    compilerError("Random.fillRandom is only defined for numeric arrays");
+  proc fillRandom(ref arr: [], seed: int) {
+    compilerError("'fillRandom' does not support non-rectangular arrays");
   }
 
-  /*
+  @chpldoc.nodoc
+  proc fillRandom(ref arr: []) {
+    compilerError("'fillRandom' does not support non-rectangular arrays");
+  }
 
-    Fills a rectangular array of numeric elements with pseudorandom values
-    in the range [`min`, `max`] (inclusive) in parallel using
-    a new :class:`PCGRandom.PCGRandomStream`  created
-    specifically for this call.  The first `arr.size` values from the stream
+  pragma "last resort"
+  @chpldoc.nodoc
+  @deprecated("The overload of `fillRandom` that accepts an 'algorithm' argument is deprecated; please remove the 'algorithm' argument")
+  proc fillRandom(ref arr: [], seed: int(64), param algorithm=_defaultRNG) {
+    compilerError("Random.fillRandom is only defined for rectangular numeric arrays");
+  }
+
+  /*
+    Fill a rectangular array of numeric values with pseudorandom values in
+    the range [``min``, ``max``] (inclusive) in parallel using a new
+    :record:`randomStream`. The first `arr.size` values from the stream
     will be assigned to the array's elements in row-major order. The
     parallelization strategy is determined by the array.
 
-    :arg arr: The array to be filled, where T is a primitive numeric type. Only
-      rectangular arrays are supported currently.
-    :type arr: `[] T`
-
-    :arg min: The (inclusive) lower bound for the random values used.
-
-    :arg max: The (inclusive) upper bound for the random values used.
+    :arg arr: An array of numeric values
+    :arg min: The (inclusive) lower bound for the random values
+    :arg max: The (inclusive) upper bound for the random values
+    :arg seed: The seed to use when creating the ``randomStream``
+  */
+  proc fillRandom(ref arr: [] ?t, min: arr.eltType, max: arr.eltType, seed: int)
```

Comment:
IMO it would read better to use `t` instead of `arr.eltType` for `min` and `max` type declarations, assuming you do not run into an issue when doing so.